### PR TITLE
refactor(nakama): break persona into a subpackage

### PR DESCRIPTION
**Pull request by namrapatel**
Closes: WORLD-738

## Overview

The purpose of this PR is to break `persona` into a sub-package that isolates all persona related functionality and helpers. Several other sub-packages were created alongside this change in order to untangle persona related dependencies.

## Brief Changelog

Refactored code into the following new sub-packages: `persona`, `pk`, `testutils`, `dispatcher`, `constants`

## Testing and Verifying

No new tests are required, all existing tests are passing.

### CODE REVIEW

**Comment by jerargus** [comment]
Don't submit this yet, I want to review it.

File: relay/nakama/pk/private_key.go
Comment (smsunarto) [inline relay/nakama/pk/private_key.go:1]: nit: rename this to signer

File: relay/nakama/constants/constants.go
[the remark this answers is not in the document]
Comment (technicallyty) [reply relay/nakama/constants/constants.go:1]: I think its better to opt for whatever makes the API look cleaner. Go itself doesn't follow this convention too strictly:

see go std lib packages: "bytes", "errors", "maps", "slices", "strings"..

[the remark this answers is not in the document]
Comment (namrapatel) [reply relay/nakama/constants/constants.go:1]: I'm also in favour of `constants`, to me `constant` just sounds off and I've never encountered this singularity pattern before

[the remark this answers is not in the document]
Comment (technicallyty) [reply relay/nakama/constants/constants.go:1]: looking further into this, i think the `constant(s)` package name might just not be a good name. https://go.dev/blog/package-names

it's a bit unclear what the package would provide. for example if it only had keys related to nakam's DB actions, probably better to name it dbkey or something like that. but constant(s) feels all a bit too vague

[the remark this answers is not in the document]
Comment (namrapatel) [reply relay/nakama/constants/constants.go:1]: I personally think that `constants` is a good and clear name, when I read that I know what the author stores in that package and why. Ofc the real problem I'm trying to solve is that several different packages depend on the stuff in constants so it cannot be in `main`. If anyone has another solution to that problem, I'm open to that, also open to renaming but I personally don't have a better name in mind after spending a few mins thinking on this. 

I vote we go forth as is, with `constants`, just as we do with `types`

File: relay/nakama/persona/persona_tag_verifier.go [hunk 1]
```diff
@@ -36,34 +37,34 @@ type txHashAndUserID struct {
...
+func (p *Verifier) AddPendingPersonaTag(userID, txHash string) {
 	p.pendingCh <- txHashAndUserID{
 		userID: userID,
 		txHash: txHash,
 	}
 }
 
-func initPersonaTagVerifier(logger runtime.Logger, nk runtime.NakamaModule, rd *receiptsDispatcher,
-) *personaTagVerifier {
+func InitPersonaTagVerifier(logger runtime.Logger, nk runtime.NakamaModule, rd *dispatcher.ReceiptsDispatcher,
```
[the remark this answers is not in the document]
Comment (namrapatel) [reply relay/nakama/persona/persona_tag_verifier.go:47]: Hm, I'm unsure what you mean here?

[the remark this answers is not in the document]
Comment (jerargus) [reply relay/nakama/persona/persona_tag_verifier.go:47]: This method should be called "NewPersonaTagVerifier", or better yet "NewVerifier" to match the go convention of creating things.

File: relay/nakama/constants/constants.go
[the remark this answers is not in the document]
Comment (technicallyty) [reply relay/nakama/constants/constants.go:1]: let's just start with moving things that don't really need to be in there, and then maybe we can get a clearer picture of what should go in here

`PersonaTagKey` is only ever used in the persona package. We should move it there and let persona manage and access its own dbkeys.

same with Cardinal collection, seems like its just related to persona tag storage.

all the endpoints and env keys are only ever used in one place, so don't think we need it in this package

the only vars that get touched by other packages are the CardinalAddress and the Namepsace

[the remark this answers is not in the document]
Comment (namrapatel) [reply relay/nakama/constants/constants.go:1]: Just made these changes in the latest commit, I'm personally still in favour of leaving the constants package in cuz it felt cleaner and more future proof. I feel like generally following this pattern of passing the var down in functions will increase complexity (LOC increased by a marginal amount and several functions now have 4+ params in latest commit).

But maybe I'm not understanding something about the future ramifications about leaving `constants` in, so I'm happy to leave constants out for now! Lmk if this is good now

**Review by jerargus — APPROVED** [review]

**Review by technicallyty — APPROVED** [review]

**Pull request merged**